### PR TITLE
Io.GitHub.donald kellett.csnaketerm

### PR DESCRIPTION
**Pull request by DonaldKellett**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions, apart from `xdg-config` for storing per-user highscores
- [x] All assets referenced in the manifest are redistributable by any party
- [x] I am the upstream author of the project
- [x] I used a GitHub domain which has a policy for delegating subdomains
- [x] The only patch in my project is `run.sh` which invokes the upstream program with a modified HOME variable, and is therefore not submitted upstream

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.DonaldKellett.csnaketerm.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        path: io.github.DonaldKellett.csnaketerm.metainfo.xml
+  - name: run.sh
+    buildsystem: simple
+    build-commands:
+      - install -D run.sh /app/bin/run.sh
+    sources:
+      - type: file
+        path: run.sh
+finish-args:
+  - '--filesystem=xdg-config'
```
Comment (barthalion) [inline io.github.DonaldKellett.csnaketerm.yml:41]: The `$XDG_CONFIG_HOME` environment variable points to writable location by default.

File: io.github.DonaldKellett.csnaketerm.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+app-id: io.github.DonaldKellett.csnaketerm
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: run.sh
+modules:
+  - 'shared-modules/python2.7/python-2.7.json'
```
Comment (barthalion) [inline io.github.DonaldKellett.csnaketerm.yml:7]: You should copy this file to your local repo and add `"cleanup": ["*"]`.

File: io.github.DonaldKellett.csnaketerm.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    sources:
+      - type: file
+        path: io.github.DonaldKellett.csnaketerm.metainfo.xml
+  - name: run.sh
+    buildsystem: simple
+    build-commands:
+      - install -D run.sh /app/bin/run.sh
+    sources:
+      - type: file
+        path: run.sh
```
Comment (barthalion) [inline io.github.DonaldKellett.csnaketerm.yml:39]: There's no need to make these separate modules, you can merge it with `csnaketerm`.

File: io.github.DonaldKellett.csnaketerm.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline io.github.DonaldKellett.csnaketerm.metainfo.xml:1]: Ideally you should maintain this file in your GitHub repo.

File: ClassicSnakeTerminal.desktop [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
Comment (barthalion) [inline ClassicSnakeTerminal.desktop:1]: Ideally you should maintain this file in your GitHub repo.

**Comment by barthalion** [comment]
bot, build io.github.DonaldKellett.csnaketerm

File: io.github.DonaldKellett.csnaketerm.yml (hunk 1 above)
Comment (DonaldKellett) [reply io.github.DonaldKellett.csnaketerm.yml:41]: This is intended, as the game saves the updated high scores in a file under `$XDG_CONFIG_HOME` on normal exit.

File: io.github.DonaldKellett.csnaketerm.yml (hunk 2 above)
Comment (DonaldKellett) [reply io.github.DonaldKellett.csnaketerm.yml:7]: Noted. Do I add the cleanup field at the top level of the YAML file?

**Comment by DonaldKellett** [comment]
Thanks for reviewing my app. A few questions:

1. Regarding merging certain modules together with `csnaketerm`, I have already made a `0.2.0` point release for the game and do not wish to modify this release or use a different commit / make yet another release just to accommodate for Flathub. Is it an absolute requirement to merge them together in order to get the app accepted into Flathub?
1. I see that the build has failed on the 128x128 image but the image has the correct dimensions as far as I can tell. Are there any extra criteria that the 128x128 image has to pass, and any way to conduct these tests locally? Apologies if the information is already included in the (failing) build output as I'm having a hard time deciphering it.

**Comment by nedrichards** [comment]
It's not an absolute requirement to modify merge the files to get the app in - it's nice if you've merged it upstream and it'll get into the next release but at the end of the day it's your choice.

For the 128x128 file - in your local build is there a `app/files/share/icons/hicolor/128x128/apps/io.github.DonaldKellett.csnaketerm.png` file? That's what it's checking for (since that file is used on the website and other software stores to represent your app)

**Comment by DonaldKellett** [comment]
I've just placed my 128x128 image under `app/files/share/app-info/icons/flatpak/128x128` and renamed it to `io.github.DonaldKellett.csnaketerm.png` as well as added a 64x64 image. As for the multiple modules, I have merged them all into a single `csnaketerm` module except Python and Node.js. Let's hope it passes the build this time ...

**Comment by DonaldKellett** [comment]
I think I previously misunderstood the part about app icons, it should be fixed now. BTW where should the `.desktop` file go in the app?

bot, build io.github.DonaldKellett.csnaketerm (can I actually do this? X-)

**Comment by DonaldKellett** [comment]
Ah, now I'm starting to understand the docs a bit better - the desktop file should be added correctly now.

bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
@nedrichards I'm having a hard time figuring out how to get the desktop icon to work, mind giving me a push in the right direction? According to my understanding of https://docs.flatpak.org/en/latest/conventions.html?highlight=.desktop#desktop-files , I placed `io.github.DonaldKellett.csnaketerm.desktop` under `/app/share/applications` and `io.github.DonaldKellett.csnaketerm.png` under `/app/share/icons/hicolor/128x128/apps`, but I still get two warnings in the test builds and the desktop icon does not display properly (though the desktop entry itself appears in the application menu).

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
I'm getting quite tired of this by now ... :tired_face: 

bot, build io.github.DonaldKellett.csnaketerm

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm ...

**Comment by DonaldKellett** [comment]
bot, build io.github.DonaldKellett.csnaketerm ... (this is getting ridiculous)

File: io.github.DonaldKellett.csnaketerm.yml (hunk 1 above)
Comment (bilelmoussaoui) [reply io.github.DonaldKellett.csnaketerm.yml:47]: What @barthalion meant here, is that you should have access to that directory already if you're not hardcoding the path and using whatever is defined in $XDG_CONFIG_HOME

**Comment by DonaldKellett** [comment]
@barthalion @nedrichards What am I doing wrong? I keep getting a build error saying that it can't open the icon file since it's a read-only filesystem. Mind pointing me in the right direction?

Anyway, I'll keep trying until the end of today. If this still doesn't work by then, I'll just close the PR since my patience is frankly running out.

**Comment by DonaldKellett** [comment]
Closing this PR since the issue remains unresolved and there is no clear path to a solution. Classic Snake Terminal will not be made available on Flathub; an entry in the Snap store should be sufficient to deliver the app to the majority of Linux users.

**Comment by barthalion** [comment]
Not sure why you closed it instead of waiting for someone to take a look. Let us know if you change your mind.

**Pull request closed without merging**